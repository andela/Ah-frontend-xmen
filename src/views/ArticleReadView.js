import React from 'react';
import { connect } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import ArticlComponent from '../components/Articles/ArticleComponent';
import getSingleArticle from '../actions/articleActions/getSingleArticle';
import CommentView from './commentView/commentView';


function parseShareLinks(article, slug) {
  const articleUrl = `${process.env.FRONTEND_BASE_URL}/articles/${slug}`;
  const shareBody = `This is an interesting article, ${article.title} ${articleUrl}`;
  const mailshare = `mailto:?Subject=${encodeURI(article.title)}&body=${shareBody}`;
  const fbshare = `https://www.facebook.com/sharer/sharer.php?u=${encodeURI(articleUrl)}`;
  const gpshare = `https://plus.google.com/share?url=${encodeURI(articleUrl)}`;
  const twshare = `https://twitter.com/home?status=${encodeURI(shareBody)}`;

  return {
    mailshare, fbshare, gpshare, twshare,
  };
}


export class ArticleReadView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      notFound: false,
      loading: true,

    };
  }


  componentWillMount() {
    const { props } = this;
    const { slug } = props.match.params;
    props.getSingleArticle(slug);
  }

  componentWillReceiveProps(props) {
    this.setState({
      loading: false,
    });
  }

  render() {
    const { article } = this.props;
    if (this.state.loading) {
      return (
        <div className="h-100 w-100 p-5 m-5  text-center">
          <span style={{ fontSize: '400%' }}>Loading.....</span>
        </div>
      );
    }

    if (article.title === '') {
      return (
        <div className="h-100 w-100 p-5 m-5  text-center">
          <span style={{ fontSize: '400%' }}>404</span>
        </div>
      );
    }

    article.share_links = parseShareLinks(article, this.props.match.params.slug);

    return (
      <div>
        <ArticlComponent
          slug={this.props.match.params.slug}
          {...article}
        />
        <CommentView />
        <ToastContainer />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  article: state.articleReducer.article,
  loading: state.articleReducer.loading,
});
export default connect(mapStateToProps, { getSingleArticle })(ArticleReadView);
