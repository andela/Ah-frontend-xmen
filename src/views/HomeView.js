import React from 'react';
import App from '../components/LandingPage/Banner';
import Articles from '../components/LandingPage/Articles';

const HomeView = () => (
  <div>
    <App />
    <Articles />
  </div>
);

export default HomeView;
