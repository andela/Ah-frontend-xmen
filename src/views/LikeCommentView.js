import React from 'react';
import { connect } from 'react-redux';
import likeCommentAction from '../actions/commentActions/commentActions';
import CommentBanner from '../components/comments/commentBanner';


class LikeCommentView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLiked: false,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick=(e) =>{
    console.log('hey');
  }

  render() {
    return (
      <CommentBanner onClick={this.handleClick} />
    );
  }
}
const mapStateToProps = state => ({
  likeCommentsAction: state.likeCommentReducer,
});
export default connect(mapStateToProps, { likeCommentAction })(LikeCommentView);
