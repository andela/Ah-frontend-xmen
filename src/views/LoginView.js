import React from 'react';
import Login from '../components/Login';
import SocialAuthView from './socialAuthView/socialAuthView';

const LoginView = props => (
  <div>
    <SocialAuthView />
    <Login />

  </div>
);

export default LoginView;
