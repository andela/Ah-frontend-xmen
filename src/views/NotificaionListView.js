
import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import notificationsAction from '../actions/notificationsAction';


export class NotificationListView extends React.Component {
  render() {
    const notifications = this.props.notifications.map((notification, index) => {
      const iconClass = notification.isUnread ? 'fas fa-circle m-3' : 'far fa-circle m-3';

      return (
        <Link to={notification.actionLink}>
          <li className="list-group-item " key={index}>
            <i className={iconClass} />
            {notification.message}
          </li>
        </Link>

      );
    });

    return (
      <div className="note-list container p-3">
        <div className="display-3">
                All notifications
        </div>
        <hr />
        <ul>
          {notifications}
        </ul>
      </div>
    );
  }
}


export function mapStateToProps(state) {
  return {
    auth: state.Auth,
    notifications: state.notificationsReducer.notifications,
    hasUnread: state.notificationsReducer.hasUnread,
  };
}

export default connect(
  mapStateToProps,
  {
    getNotifications: notificationsAction,
    markAsRead: notificationsAction,
  },
)(NotificationListView);
