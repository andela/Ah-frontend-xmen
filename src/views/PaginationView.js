/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Pagination from '../components/Pagination/Pagination';
import fetchArticlesAction from '../actions/articleActions/fetchArticlesAction';

export class Paginations extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageNumbers: [],
      allArticles: 7,
      currentPage: 1,
      totalPages: 1,
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.count && nextProps.count > 1) {
      if (nextProps.count > 6) {
        this.setState({
          allArticles: nextProps.count,
          totalPages: nextProps.count / 6,
        });
      } else {
        this.setState({ allArticles: nextProps.count, totalPages: 1 });
      }
    }
  }

  handleNext = (nextPageIndex) => {
    if (nextPageIndex < +this.state.totalPages && nextPageIndex >= 1) {
      this.props.fetchArticlesAction(nextPageIndex);
      this.setState({ currentPage: nextPageIndex });
    }
  };

  handlePrevious = () => {
    this.props.fetchArticlesAction(this.state.currentPage - 1);
  };

  render() {
    const { currentPage, totalPages } = this.state;
    let { pageNumbers } = this.state;
    pageNumbers = [];

    for (let i = 1; i <= totalPages; i += 1) {
      pageNumbers.push(i);
    }
    return (
      <Pagination
        pageNumbers={pageNumbers}
        nextPage={this.handleNext}
        previousPage={this.handlePrevious}
        currentPage={currentPage}
      />
    );
  }
}

export const mapStateToProps = state => ({
  newArticles: state.articles.articles,
  count: state.articles.allArticles,
});

export default connect(
  mapStateToProps,
  {
    fetchArticlesAction,
  },
)(Paginations);
