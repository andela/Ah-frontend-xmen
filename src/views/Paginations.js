import React, { Component } from 'react';
import { connect } from 'react-redux';
import Pagination from '../components/Pagination/Pagination';
import {
  fetchArticlesAction,
  getAllArticlesAction,
} from '../actions/articleActions/fetchArticlesAction';

class Paginations extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageNumbers: [],
      allArticles: 7,
      currentPage: 1,
    };
  }

  componentWillReceiveProps(nextProps) {
    console.log(nextProps);
    if (nextProps.count && nextProps.count > 1) {
      console.log(nextProps.count);

      this.setState({ allArticles: nextProps.count });
    }
  }

  handleNext = (nextPageIndex) => {
    this.props.getAllArticlesAction(nextPageIndex);
    this.setState({ currentPage: nextPageIndex });
  };

  handlePrevious = () => {
    this.props.getAllArticlesAction(this.state.currentPage - 1);
  };

  render() {
    const { allArticles, currentPage } = this.state;
    let { pageNumbers } = this.state;
    pageNumbers = [];
    let totalPages;
    if (allArticles > 6) {
      totalPages = allArticles / 6;
    }

    for (let i = 1; i <= totalPages; i++) {
      pageNumbers.push(i);
    }
    return (
      <Pagination
        pageNumbers={pageNumbers}
        nextPage={this.handleNext}
        previousPage={this.handlePrevious}
        currentPage={currentPage}
      />
    );
  }
}

export const mapStateToProps = state => ({
  newArticles: state.articles.articles,
  count: state.articles.allArticles,
});

export default connect(
  mapStateToProps,
  {
    fetchArticlesAction,
    getAllArticlesAction,
  },
)(Paginations);
