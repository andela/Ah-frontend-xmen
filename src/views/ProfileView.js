/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { connect } from 'react-redux';
import fetchProfile from '../actions/ProfileActions';
import Profile from '../components/Profile';

export class ProfileView extends React.Component {
  constructor(props, { match }) {
    super(props, { match });
    this.componentDidMount = this.componentDidMount.bind(this);
  }

  componentDidMount() {
    this.props.getProfiles(this.props.match.params.username);
  }

  render() {
    if (this.props.error) {
      return (
        <div className="errormsg card">
          {' '}
    Error!
          {' '}
          {this.props.error.message}
&nbsp;
    Please Login
        </div>
      );
    }
    if (this.props.loading) {
      return <div className="loadingmsg card">Loading...</div>;
    }
    const ownUsername = localStorage.getItem('username');
    const paramUsername = this.props.match.params.username || ownUsername;
    const isOwnProfile = ownUsername === paramUsername;
    return (
      <div>
        <Profile profile={this.props.profile} isOwnProfile={isOwnProfile} />
      </div>
    );
  }
}


const mapStateToProps = state => ({
  profile: state.profileReducer.profile,
  loading: state.profileReducer.loading,
  error: state.profileReducer.error,
});


const mapDispatchToProps = dispatch => ({
  getProfiles: (username) => {
    dispatch(fetchProfile(username));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(ProfileView);
