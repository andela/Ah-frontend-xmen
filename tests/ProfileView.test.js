import React from 'react';
import { shallow } from 'enzyme';
import { ProfileView } from '../src/views/ProfileView';


describe('profile view', () => {
  it('should render with loading', () => {
    const props = {
      getProfiles: jest.fn(),
      loading: 'loading',
      match: { params: {} },
    }; const wrapper = shallow(<ProfileView {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render witho errors', () => {
    const props = {
      getProfiles: jest.fn(),
      error: 'error',
      match: { params: {} },
    }; const wrapper = shallow(<ProfileView {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    const props = {
      getProfiles: jest.fn(),
      match: { params: {} },
    }; const wrapper = shallow(<ProfileView {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
