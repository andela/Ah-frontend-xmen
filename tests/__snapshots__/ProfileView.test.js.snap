// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile view should render with errors 1`] = `ShallowWrapper {}`;

exports[`profile view should render with loading 1`] = `ShallowWrapper {}`;

exports[`profile view should render without crashing 1`] = `ShallowWrapper {}`;
